### ISSUE TITLE
Difference between eigs() and eigen()

### ISSUE THREAD

**Issue by pbruneau** [yixuan/rARPACK#6]
I recently observed a difference between the outputs of the classic eigen() function, and your Arnoldi variant eigs() that extracts only the few first eigenpairs. Here is some sample code illustrating the problem:

```
library(rARPACK)
library(speccalt)

# compute kernel matrix from rows of 'synth5' data set
# then its Laplacian
kern <- local.rbfdot(synth5)
diag(kern) <- 0
deg <- sapply(1:(dim(synth5)[1]), function(i) {
return(sum(kern[i,]))
})
L <- diag(1/sqrt(deg)) %*% kern %*% diag(1/sqrt(deg))

eig1 <- eigs(L, 6)
eig2 <- eigen(L, symmetric=TRUE)
```

eig1$values then reads:
1.0000000 1.0000000 0.9993805 0.9992561 0.9985084 0.9975311

whereas eig2$values reads:
1.0000000 1.0000000 1.0000000 1.0000000 0.9993805 0.9992561
which is the correct result (eigenvalue 1 has multiplicity 4 in that example).

I guess there is an issue between Arnoldi methods and eigenvalues with multiplicities greater than 1 (indeed at the end of the series the unique eigenvals look identical)... The Matlab version of eigs() does not seem affected though - I tried to csvwrite and read the L matrix from R to Matlab, run:
```
eig1 = eigs(L, 6)
````

and the eigen() result was output then.

I would gladly use and quote your package, as the speedup is very useful to me, but this issue is blocking in my case...
Please feel free to ask any supplementary information, and best regards,
Pierrick

**Comment by yixuan** [comment]
Hello @pbruneau ,
Thanks for reporting this. I think you are correct. In general eigen solvers are hard to detect eigenvalues with multiplicity more than 1. This is even harder for Arnoldi method since it only calculates and sorts part of the eigenvalues.
Also, the results are subject to precision tolarence. For your example, you can get the same results as `eigen()` by specifying a smaller `tol` parameter, as the example below shows:
```
eigs(L, 6, opts = list(tol = 1e-17))
```
However, it greatly increases computation time, and I don't have a criterion to tell how small `tol` is enough.

**Comment by yixuan** [comment]
Another way is to increase `ncv`, which is somewhat more "mysterious".
```
eigs(L, 6, opts = list(ncv = 300))
```

**Comment by pbruneau** [comment]
Hi @yixuan,

Thanks for the suggestions! I tried them out, with success - but at a (computational) cost indeed.

After several trials, I think I'll go for adjusting tol (1e-16 appeared as sufficient, in the same range of R double precision BTW).

Also, for the record, I tried out with larger matrices (up to 4000x4000), and the benefit of extracting some constant number of eigenvalues with eigs() is still very significant then.

Thank for the tips,
Pierrick
